app/models/decision.rb

```diff
@@ -4,14 +4,17 @@ class Decision < ApplicationRecord
 
   validates :result, :created_by, presence: {message: "Make a decision to approve or reject the claim"}
   validate :claim_must_be_approvable, if: :approved?, on: :create
+  validate :claim_must_have_undoable_decision, if: :undone?, on: :update
+
+  scope :not_undone, -> { where(undone: false) }
```

Comment:
I wrestled with the naming of this scope a little, could we avoid the double negative somehow? Perhaps something like `:active` would be better?